database/gdb/gdb.go

```diff
@@ -708,7 +708,7 @@ const (
 	ctxKeyCatchSQL            gctx.StrKey = `CtxKeyCatchSQL`
 	ctxKeyInternalProducedSQL gctx.StrKey = `CtxKeyInternalProducedSQL`
 
-	linkPattern            = `(\w+):([\w\-\$]*):(.*?)@(\w+?)\((.+?)\)/{0,1}([^\?]*)\?{0,1}(.*)`
+	linkPattern            = `(\w+):([^:]*):(.*?)@(\w+?)\((.+?)\)/{0,1}([^\?]*)\?{0,1}(.*)`
```

Comment:
- It would be better using `^(\w+):(.*?):(.*?)@(\w+?)\((.+?)\)/{0,1}([^\?]*)\?{0,1}(.*?)$` for the pattern.
- Necessary unit testing case should be added for this update here: https://github.com/gogf/gf/blob/92eab81926dea3ef090905c8f4da267c84fd11a3/database/gdb/gdb_z_mysql_internal_test.go#L297

Reply:
ok, i open a new pr #4064